src/core/qgspathresolver.cpp

```diff
@@ -223,12 +223,12 @@ QString QgsPathResolver::setPathWriter( const std::function<QString( const QStri
 
 bool QgsPathResolver::removePathWriter( const QString &id )
 {
-  const size_t prevCount = sCustomWriters->size();
-  sCustomWriters()->erase( std::remove_if( sCustomWriters->begin(), sCustomWriters->end(), [id]( std::pair< QString, std::function< QString( const QString & ) > > &a )
+  const size_t prevCount = sCustomWriters()->size();
```

Comment:
[operator()](https://codebrowser.dev/qt5/qtbase/src/corelib/global/qglobalstatic.h.html#_ZN13QGlobalStaticclEv) is interesting if we test the returned pointer but here, it would just crash if it's nullptr. At last with previous code, get we get the assert on debug mode.

Reply:
I don't believe it's possible for this to be nullptr. Q_GLOBAL_STATIC ensures that it's created.